corehq/apps/api/resources/pagination.py

```diff
@@ -60,3 +65,31 @@ def page(self):
             self.collection_name: self.objects,
             'meta': meta,
         }
+
+
+def response_for_cursor_based_pagination(query, request_params, datasource_adapter):
+    """Creates a response dictionary that can be used for cursor based pagination
+     :returns: The response dictionary
+    """
+    records = get_datasource_records(query, datasource_adapter)
+    return {
+        "objects": records,
+        "meta": {
+            "next": _get_next_url_params(records),
+            "limit": request_params["limit"]
+        }
+    }
+
+
+def _get_next_url_params(datasource_records):
+    """ Constructs the query string containing a base64-encoded cursor that points to the last entry in
+    `datasource_records`
+    :returns: The query string"""
+    if not datasource_records:
+        return None
+
+    last_object = datasource_records[-1]
+    cursor_params = {"last_doc_id": last_object["doc_id"], "last_inserted_at": last_object["inserted_at"]}
+    encoded_cursor = b64encode(urlencode(cursor_params).encode('utf-8'))
+    next_params = {'cursor': encoded_cursor}
+    return f'?{urlencode(next_params)}'
```

Comment:
That `?` looks fragile, because it assumes that these will be the only URL params. Is that definitely always going to be true?  Is there a reason we don't just return `urlencode(next_params)` and leave it up to the caller to construct the URL? (See my other comment for a possible way to make this more robust.)

Reply:
https://github.com/dimagi/commcare-hq/pull/33380/commits/fc4d78578c210e1bf5ec978ac414f70733218816